File: miscellaneous/sorting/insertion-sort/20221231/insertion_sort.py
from typing import List


def insertion_sort(nums: List[int]) -> List[int]:
    for i in range(1, len(nums)):
        current = nums[i]
        j = i - 1

        while current < nums[j] and j >= 0:
            nums[j+1] = nums[j]
            j -= 1

        j += 1
        nums[j] = current

    return nums

File: miscellaneous/sorting/insertion-sort/20221231/insertion-sort.php
<?php

declare(strict_types=1);

function insertionSort(array $nums): array {
    for ($i = 1; $i < count($nums); ++$i) {
        $j = $i - 1;
        $current = $nums[$i];

        while ($j >= 0 && $current < $nums[$j]) {
            $nums[$j+1] = $nums[$j--];
        }

        $nums[++$j] = $current;
    }

    return $nums;
}